### PR TITLE
vzic: init at 0-unstable-2024-06-04

### PR DESCRIPTION
**Pull request by Moraxyc**
## Description of changes

A program to convert the IANA (formerly Olson) timezone database files into VTIMEZONE files compatible with the iCalendar specification (RFC2445).

This package is required by cyrus-imapd and libical.

Also provide vtimezone files in share/vtimezone directly.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/vz/vzic/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/vz/vzic/package.nix:40]: Typo in url
```suggestion
    homepage = "https://github.com/libical/vzic";
```

File: pkgs/by-name/vz/vzic/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/vz/vzic/package.nix:31]: The Makefile in `vzic` uses [conditional assignment](https://www.gnu.org/software/make/manual/html_node/Conditional-Assignment.html). This means that they can simply be overridden by environment, avoiding patching the Makefile.

```suggestion
  OLSON_DIR = version;
  PRODUCT_ID = "-//NixOS//NONSGML Citadel calendar//EN";
  TZID_PREFIX = "/NixOS/Olson_%D_1/";
```

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/vz/vzic/package.nix
Comment (Sigmanificient) [inline pkgs/by-name/vz/vzic/package.nix:44]: Please add yourself within the `meta.maintainers` section

**Review by Sigmanificient — CHANGES_REQUESTED** [review]

File: pkgs/by-name/vz/vzic/package.nix
Comment (Moraxyc) [reply pkgs/by-name/vz/vzic/package.nix:44]: Sadly, the PR haven’t been merged yet.
https://github.com/NixOS/nixpkgs/pull/299069

Comment (Sigmanificient) [reply pkgs/by-name/vz/vzic/package.nix:42]: Usually, people had themselves when they port their first package

Comment (Moraxyc) [reply pkgs/by-name/vz/vzic/package.nix:42]: The same
https://github.com/NixOS/nixpkgs/pull/281278
:(

Comment (Sigmanificient) [reply pkgs/by-name/vz/vzic/package.nix:42]: Oh well, I'm sorry about it

Comment (Sigmanificient) [reply pkgs/by-name/vz/vzic/package.nix:42]: I can only approve the change for your maintainer pr, hopefully it will help.

Comment (Moraxyc) [reply pkgs/by-name/vz/vzic/package.nix:42]: Thx for approval!

**Comment by Sigmanificient** [comment]
Result of `nixpkgs-review pr 299093` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vzic</li>
  </ul>
</details>

**Review by Sigmanificient — CHANGES_REQUESTED** [review]
I saw that the license attribute was not defined, checked the repository, didn't find any license file. Looked into the [pkgs readme](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), because I wasn't sure about it. Here was it's says:

> `meta.license` must be set and match the upstream license.
If there is no upstream license, `meta.license` should default to `lib.licenses.unfree`.
If in doubt, try to contact the upstream developers for clarification.

**Comment by Moraxyc** [comment]
According to the issue https://github.com/libical/vzic/issues/8 , vzic may be published under GPL license. But there's no response by libical group. I'd like to request clarification to them then.

**Comment by Moraxyc** [comment]
https://github.com/libical/vzic/blob/master/LICENSE.txt
Solved license problem.

I commented maintainer section until my first PR has been merged.

**Comment by Moraxyc** [comment]
Result of `nixpkgs-review pr 299093` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vzic</li>
  </ul>
</details>

File: pkgs/by-name/vz/vzic/package.nix
Comment (Moraxyc) [reply pkgs/by-name/vz/vzic/package.nix:34]: There's no install rule in Makefile, so it should be done manually

Comment (AndersonTorres) [reply pkgs/by-name/vz/vzic/package.nix:34]: the `runHook  preInstall` and `runHook  postInstall` are still mandatory.

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

File: pkgs/by-name/vz/vzic/package.nix
Comment (OPNA2608) [inline pkgs/by-name/vz/vzic/package.nix:3]:
```suggestion
  lib,
  buildPackages,
```

File: pkgs/by-name/vz/vzic/package.nix
Comment (OPNA2608) [inline pkgs/by-name/vz/vzic/package.nix:37]: When cross-compiling, you can't natively execute things you built during a derivation.

```suggestion
    ${stdenv.hostPlatform.emulator buildPackages} ./vzic --output-dir $out/share/vtimezone
```

File: pkgs/by-name/vz/vzic/package.nix
Comment (OPNA2608) [inline pkgs/by-name/vz/vzic/package.nix:50]: If we need to execute what we built, then we'll need to have an emulator available. If there is none, then the derivation cannot be built.

```suggestion
    broken = !stdenv.hostPlatform.emulatorAvailable buildPackages;
  };
```

File: pkgs/by-name/vz/vzic/package.nix
Comment (OPNA2608) [inline pkgs/by-name/vz/vzic/package.nix:26]: These dependencies are not split correctly. (`strictDeps = true` could enforce proper separation here)

- pkg-config is invoked at built time, so it needs to go to `nativeBuildInputs`
- glib gets compiled & linked against, so it needs to go to `buildInputs`

tzdata & libical don't seem to make any difference on the build result. I see that libical at least is present in upstream's hardcoded FHS include flags, so maybe it's included in case it's ever needed? Not sure about tzdata, please decide on your own if it's really needed & which input it needs to go into - I'll leave it as-is for the suggestion.

The Makefile also uses a hardcoded `pkg-config`, which we don't always have.
```suggestion
  postPatch = ''
    substituteInPlace Makefile \
      --replace-fail 'pkg-config' "$PKG_CONFIG"
  '';

  nativeBuildInputs = [
    pkg-config
    tzdata
  ];

  buildInputs = [
    glib
    libical
  ];
```

**Review by OPNA2608 — CHANGES_REQUESTED** [review]

File: pkgs/by-name/vz/vzic/package.nix
Comment (OPNA2608) [inline pkgs/by-name/vz/vzic/package.nix:12]: This should denote some unique-ish identifier of upstream's version. As-is, this only says which `tzdata` directory will be used for building. That's insufficient, because it's present in multiple commits since its introduction.

```suggestion
  version = "0-unstable-2024-03-09";
```

I think if you want the information to be present in the name, maybe it should be added as a postfix to the `pname`? I.e. `vzic-tzdata2024a`. version is the wrong place, *maybe* unless you pin `src.rev` to the commit that introduced tzdata2024a: https://github.com/libical/vzic/commit/7ff0ce05c32ad3218cd12f71e6194dd1f7add299

File: pkgs/by-name/vz/vzic/package.nix
Comment (OPNA2608) [inline pkgs/by-name/vz/vzic/package.nix:28]: See note about `version`.

```suggestion
    OLSON_DIR = "tzdata2024a";
```

**Review by OPNA2608 — DISMISSED** [review]
Missed the version.

File: pkgs/by-name/vz/vzic/package.nix
Comment (Moraxyc) [reply pkgs/by-name/vz/vzic/package.nix:12]: I prefer to pin the rev to that commit. Putting the tzdata version in pname seems to be weird.

Comment (drupol) [reply pkgs/by-name/vz/vzic/package.nix:12]: I think you should use what @OPNA2608 advised here.

**Review by drupol — APPROVED** [review]
Thank you

**Review by AndersonTorres — APPROVED** [review]
Result of `nixpkgs-review pr 299093` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>vzic</li>
  </ul>
</details>

File: pkgs/by-name/vz/vzic/package.nix
Comment (AndersonTorres) [inline pkgs/by-name/vz/vzic/package.nix:23]: P.S.:  it should be upstreamed

**Pull request merged**
